### PR TITLE
Add validation to item creation and require authentication

### PR DESCRIPTION
**Pull request by betrisey**
- Add validation for title (cannot be empty), errors are displayed on the EditText
- Use [Material Design text fields](https://material.io/components/text-fields)
- Merged create and edit buttons to simplify the code
- Refactored NewEditFragment
- Require user authentication to create an item

https://user-images.githubusercontent.com/1939395/118702508-4d8f6800-b815-11eb-887c-bb81d93cf9e2.mp4

### CODE REVIEW

File: app/src/main/java/com/example/sharingang/ui/fragments/NewEditFragment.kt [hunk 1]
```diff
@@ -121,43 +125,72 @@ class NewEditFragment : Fragment() {
...
+     *
+     * @return whether the form is valid and ready to be saved
+     */
+    private fun validateForm(): Boolean {
+        val titleEmpty = binding.title?.trim()?.isEmpty() ?: true
+        binding.itemTitleContainer.error =
+            if (titleEmpty) getString(R.string.required_field) else null
+        if (titleEmpty) {
+            binding.itemTitle.requestFocus()
+        }
```
Comment (cronokirby) [inline app/src/main/java/com/example/sharingang/ui/fragments/NewEditFragment.kt:189]: Maybe merge these two checks?

File: app/src/main/java/com/example/sharingang/ui/fragments/NewEditFragment.kt [hunk 2]
```diff
@@ -121,43 +125,72 @@ class NewEditFragment : Fragment() {
...
     }
 
+    /**
+     * Check if the form is valid.
+     * If not, it displays error messages on the view.
+     *
+     * @return whether the form is valid and ready to be saved
+     */
+    private fun validateForm(): Boolean {
+        val titleEmpty = binding.title?.trim()?.isEmpty() ?: true
```
Comment (cronokirby) [inline app/src/main/java/com/example/sharingang/ui/fragments/NewEditFragment.kt:184]: Isn't this just `== null`?

Comment (cronokirby) [reply app/src/main/java/com/example/sharingang/ui/fragments/NewEditFragment.kt:184]: Oh, I see nvmd

**Review by cronokirby — APPROVED** [review]
Looks good to me!

**Review by bafavre — APPROVED** [review]
Nice, well done! :)

**Pull request merged**